package/Dockerfile.agent

```diff
@@ -54,26 +44,44 @@ RUN /usr/bin/unshare --version && \
     /usr/bin/nsenter --version
 
 ARG ARCH=amd64
-ENV KUBECTL_VERSION v1.27.10
+ENV KUBECTL_VERSION=v1.29.5
 ARG VERSION=${VERSION}
+ARG RANCHER_REPO=rancher
 
 RUN curl -sLf https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/${ARCH}/kubectl > /usr/bin/kubectl && \
     chmod +x /usr/bin/kubectl
 
-ENV LOGLEVEL_VERSION v0.1.6
+ENV LOGLEVEL_VERSION=v0.1.6
 
 RUN curl -sLf https://github.com/rancher/loglevel/releases/download/${LOGLEVEL_VERSION}/loglevel-${ARCH}-${LOGLEVEL_VERSION}.tar.gz | tar xvzf - -C /usr/bin
 
-LABEL io.cattle.agent true
-ENV AGENT_IMAGE rancher/rancher-agent:${VERSION}
+LABEL io.cattle.agent=true
+ENV AGENT_IMAGE=${RANCHER_REPO}/rancher-agent:${VERSION}
 # For now, this value needs to be manually synced with the one in the main Dockerfile. This pins downstream webhook's version.
 ARG CATTLE_RANCHER_WEBHOOK_VERSION
 ENV CATTLE_RANCHER_WEBHOOK_VERSION=$CATTLE_RANCHER_WEBHOOK_VERSION
 ARG CATTLE_RANCHER_PROVISIONING_CAPI_VERSION
 ENV CATTLE_RANCHER_PROVISIONING_CAPI_VERSION=$CATTLE_RANCHER_PROVISIONING_CAPI_VERSION
-ENV SSL_CERT_DIR /etc/kubernetes/ssl/certs
+ENV SSL_CERT_DIR=/etc/kubernetes/ssl/certs
 COPY --from=rancher /var/lib/rancher-data /var/lib/rancher-data
-COPY --from=rancher /usr/bin/tini /usr/bin/
+
+# Things that can't just be copied from the rancher image because the architectures might differ.
```

Comment:
This largely depends on how the `Dockerfile` is created and whether `docker buildx` is being used. Please note that the `COPY` command [can be used](https://github.com/rancher/security-scan/blob/master/package/Dockerfile#L120) to be architecture-aware.